### DISCUSSION TITLE
[NET] received packet from  on ignored interface

### DISCUSSION THREAD

**Discussion by orottler** [strongswan/strongswan#2169] [Q&A] [answered]
**System (please complete the following information):**
 - OS: RedHat9.3
 - Kernel version (if applicable): 5.14.0-362.18.1.el9_3.x86_64
 - strongSwan version(s): 5.9.13
 - Tested/confirmed with the latest version: yes (think is the latest stable)

**Describe the bug**
doesn't receive the responder packets because of
Mar 14 16:42:25 03[NET] received packet from 192.168.106.1[500]  or to <my IP>[500] on ignored interface
    my IP: 192.168.106.1[500] (load_test overloading vlan2424)
    my IP 192.168.128.235 (interface bond1 for a single connection ipsec up) 

**To Reproduce**
1. establish a point to point connection betwen client and gateway
     15: vlan2424@bond1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 30:3e:a7:04:7e:30 brd ff:ff:ff:ff:ff:ff                                                                 (lacp portchannel of em3 and em4)
    inet 192.168.106.235/24 brd 192.168.106.255 scope global noprefixroute vlan2424  (inserted by load_tester)
       valid_lft forever preferred_lft forever
    inet 192.168.128.1/17 scope global vlan2424
       valid_lft forever preferred_lft forever

2.  don't use any firewall and XFRM;
      proof:  Mar 14 16:40:56 00[KNL] received netlink error: Policy offload is not supported (22)

3. ipsec start / ipsec up <connection>

**Expected behavior**

Connections should be established/routed to the socket or give a hint what to do/ what the real problem could be

**Logs/Backtraces**
[conf-log.tar.gz](https://github.com/strongswan/strongswan/files/14605697/conf-log.tar.gz)

**Additional context**

Tried that first time since years with the old config and a new system/new build, added interface statements in strongswan.conf and swanctl.conf without any change , the last one can be seen in the log 

 Mar 14 16:40:56 00[KNL] known interfaces and IP addresses:
Mar 14 16:40:56 06[CFG]    interface = vlan2424,vlan2424@bond1,em3,em4

**Comment by tobiasbrunner** [comment, marked as answer]
> Mar 14 16:40:56 06[CFG] interface = vlan2424,vlan2424@bond1,em3,em4

While unrelated, this is definitely wrong. That option only takes a single interface name and only affects outbound IPsec policies. Not sure what your intention was there, but you might want to read [the docs](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html).

> inet 192.168.106.235/24 brd 192.168.106.255 scope global noprefixroute vlan2424  (inserted by load_tester)

No, the IP that load-tester tries to install is `192.168.128.1`, which is the first address in the subnet you configured (`192.168.128.0/17`). It even logs that this IP is already found:

> Mar 14 16:40:56 07[KNL] virtual IP 192.168.128.1 is already installed on vlan2424

Anyway, if you configure `charon.interfaces_use`, only packets received with a destination IP that's found on an interface that matches a name in that list will be accepted. This uses an internal lookup that searches for an up and unfiltered interface with that IP. That both the IP and interface is known can be seen from the log message above, however, since the list of known interfaces, which only lists up and unfiltered interfaces as well, is empty

> Mar 14 16:40:56 00[KNL] known interfaces and IP addresses:

indicates that all your interfaces are filtered (after all the RTM_NEWLINK and RTM_NEWADDR messages you see in the log there should be a list of interfaces and addresses, like e.g. seen [here](https://www.strongswan.org/testing/testresults/ikev2/net2net-rekey/moon.daemon.log)).

Do any of the files included by strongswan.conf set `charon.intefaces_use` as well?

**Reply by orottler** [reply]
I agree, this unrelated config was only some kind of panic. 
But now I got the problem, I did specify interface use in  charon.conf uppercase VLAN2424, so stupid - works now. Thanks a lot !
May be you could in newer version log the results of the filter and wich config did deploy them.
Thank you/Have a nice Weekend !

**Reply by tobiasbrunner** [reply]
> I did specify interface use in charon.conf uppercase VLAN2424

Yeah, interface names are case sensitive. It was configured lowercase in the config you uploaded, though.

> May be you could in newer version log the results of the filter and wich config did deploy them.

I guess we could log interfaces we skip when listing the known interfaces and IP addresses on level 3 (the other messages are on level 2).
